### DISCUSSION TITLE
Github will force 2FA, better to leave

### DISCUSSION THREAD

**Discussion by jackffmm** [davmac314/dinit#233] [General]
In addition to the old, well known reasons, github will soon require the phone number to login.
So more people will stop use it. I suggest to move from github to gitlab* or codeberg, or similar...

\* I dont think that gitlab is innocent and pure (but still better than gh)

**Comment by mobin-2008** [comment]
I **_personally_** prefer to host **_my own project(s)_** on somewhere else, Because

GitHub formely had a simple process to signup. Also It has shiny interfaces for more junior contributers. It offers "PR" and "fork for contribution" things which is more "user friendly" over "classic [PATCH]s through mailing lists". That was my reason to prefer GitHub over anything else, but situation get changed now: Signing up in GitHub isn't easy anymore (Setup 2FA is now required). Also I started to read the [halloween docs](http://www.catb.org/~esr/halloween/index.html) and I'm not sure changing CEO and posting _Microsoft LOVES Open Source_ did changed these horrible opinions in the depth of Microsoft.

I personally prefer Sourcehut over GitHub because:

1. It has a well-designed, without javascript and rellay fast web interface, Signing up is a simple process.
2. Its CI has a very very interesting feature, You can ssh into a failed build, It's really really useful feature instead of gussing why some builds gets fail (sometimes log doesn't tell about details).
3. It uses mailing list as "discussing place" and patches. I really enjoy times I use `git send-email` to send a patch to a mailing list
4. Its [privacy policy](https://man.sr.ht/privacy.md) is short and respect users privacy
5. It has ticket tracker for easily reporting bugs and other things, Also it support more "state" for tickets (FIXED, BY_DESIGN, WON'T FIX, NOT_OUR_BUG, INVALID, ...) (Github has the "issues" of course).

It costs 2$ per month (and it's my primary reason for don't using it, because paying by credit card is nightmare for me). Also it is in alpha stage and maybe they increase its price. Also it's incomplete-yet.

Last thing to note: It's not my personal project and it's just my personal opinion.

**Comment by mgrinzPlayer** [comment]
It won't require phone number. There are alternatives, here from email they sent you:
`Enrolling in 2FA is easy, and we accept several options including TOTP mobile apps, text messages (SMS), security keys, and GitHub Mobile.`

For you those "mobile apps" is a no-no. SMS require phone number, nope. Security keys (hardware devices) probably are not an option.
But, that statement from email is misleading. You can use TOTP desktop application. That TOTP application can be totally offline, it just must run on machine with current date and time.

You can also write your own TOTP application, e.g.  if your IDE provides HMACSHA1 and base32decode you have to write about 20 lines of code.

Edit:
I think most people will just use any "password manager" application which has TOTP feature.

**Comment by davmac314** [comment]
Here are some thoughts:

- a phone number is not actually required, as noted above
- the "old, well known reasons" could be referring to numerous things and while I probably have some idea what you're talking about, it would be better for discussion to be specific, I think
 - People who are unhappy using Github for whatever reason can still contribute: issues, patches, PRs can be sent to me directly via email.
- AFAIUI 2FA is only required for code contributions, not for eg opening issues/discussions
- Github remains, as far as I know, the most popular repo hosting service and I don't see that changing with the introduction of 2FA. Moving to another provider is likely to decrease visibility. Most traffic to the repository is generated by Github itself (followed by the website and Google which are about equal).
- In terms of general practice, I actually agree that requiring 2FA for code contributions is a good choice.
- Moving would require a bunch of admin work that I don't really have time for (move issues, and especially get CI up and running). I barely have enough time to spend on coding, I don't want to have to do that stuff (and to be blunt, with no skin in the game yourself, you don't really have the right to suggest that I should; even writing this response takes time I could spend on actually improving Dinit).
- The small amount of funding I get for Dinit is via Github sponsors. Other than a one-off, I haven't received anything else. I'm not willing to sacrifice it at this stage. I recognise that this is a self-fulfilling argument though so I will look at adding alternatives.
- I'm not 100% happy with Github but it remains the best option for now IMO (and alternatives generally don't offer the things I would like to have, such as CI configuration separate from the main repo).
- If someone wanted to mirror Dinit on an alternate hosting provider, I wouldn't be opposed.

**Comment by jackffmm** [comment]
@davmac314 
(your point) 1. the other options are worse than phone number except maybe totp ( tnx @mgrinzPlayer ), but idk how many will want to run an app in background just for gh. i dont at the moment.
2. github is owned by microsoft. Using it u train a AI that may eventually destroy the humanity in future.
3. much fewer people will do that.
4. thats good!
5. it may change... 
6. me too if the service doesnt aim at personal data. not forced anyway.
7. i suggest it for your advantage.
9. the best untill now IMO.

Anyway, not just people who care for theyr privacy will stop use it, also the lazy ones, that doesnt bother do an extra step. idk, we will see how many people will leave gh (in general).

**Comment by hholst80** [comment]
a move from github should be based on technical considerations not rms-like ideology.

**Comment by realroot2185** [comment]
There are more details of the reasons given at https://giveupgithub.org/.
